app/models/mixins/cloud_tenant_mixin.rb

```diff
@@ -1,8 +1,22 @@
 module CloudTenantMixin
   extend ActiveSupport::Concern
 
-  # TODO(lpichler) add synchronization
+  OPENSTACK_CLOUD_PROVIDER = 'Openstack Cloud Provider'.freeze
+
   def sync_cloud_tenants_with_tenants
     return unless supports_cloud_tenants?
+    sync_root_tenant
+  end
+
+  def sync_root_tenant
+    ems_tenant = source_tenant
+
+    if source_tenant.nil?
```

Comment:
It may be clearer here if you check for `ems_tenant.nil?` since it was just assigned above and is the subject of the logic that follows.